### PR TITLE
CR 21383: Integrate attachments into doppleganger mirror

### PR DESCRIPTION
**Pull request by humbletim**
#### Testing:
* Launch the new doppleganger / mirror script from this PR; rawgit link http://cdn.rawgit.com/humbletim/hifi/d6eb77e/unpublishedScripts/marketplace/doppleganger-attachments/app-doppleganger-attachments.js
* Make sure to be wearing the default Mannequin avatar.
* Verify the new tablet/toolbar button appears.
* Activate the doppleganger mirror.
* Try adding/removing/adjusting various attachments:
    * ... @ welcome (or dev-welcome) shops.
    * ... manually using the Avatar > Attachments UI dialog.
* Verify each attachment change is reflected on the doppleganger.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/863/HighFidelity-Beta-PR10743-e6e1ae16db76ef9485aa0be94db24081e39271b9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/863/HighFidelity-Beta-PR10743-e6e1ae16db76ef9485aa0be94db24081e39271b9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/998/HighFidelity-Beta-PR10743-0ef623ba069818da60463a47188488c9fe58392c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/998/HighFidelity-Beta-PR10743-0ef623ba069818da60463a47188488c9fe58392c.exe

**Comment by humbletim** [comment]
- [x] Revert the tablet icon back to what you had for Mirror Mode (and have it say "MIRROR" like before too)
- [x] Add a logAction called "doppleganger_attachments" that records the number of attachments the user has on when using the doppleganger
- [x] Commit those changes to your GitHub PR
- [x] Let me know what assets we need to swap out in the Marketplace submission.
    - **`app-doppleganger-attachments.js`** (main script; replaces `app-doppleganger.js`)
    - **`doppleganger.js`** (updated)
    - **`doppleganger-attachments.js`** (new)
    - **`model-helper.js`** (new)
    - **`utils.js`** (new)

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Launch the new doppleganger / mirror script from this PR; rawgit link http://cdn.rawgit.com/humbletim/hifi/0ef623b/unpublishedScripts/marketplace/doppleganger-attachments/app-doppleganger-attachments.js
**Pass**
Make sure to be wearing the default Mannequin avatar.
**Pass**
Verify the new tablet/toolbar button appears.
**Pass**
Activate the doppleganger mirror.
**Pass**
Try adding/removing/adjusting various attachments:
... @ welcome (or dev-welcome) shops. **Pass, Tested in Dev-Welcome**
... manually using the Avatar > Attachments UI dialog. **Pass, When moving attachments and such it makes it disappear for a very brief period while it updates location on the mirrored character, seems to work well**
Verify each attachment change is reflected on the doppleganger. **Pass, though as mentioned, during update of new position/size/rotation of attachments, it briefly disappears and re-appears to the new variables set**

All **Pass** :)

Please see image of the absolute fabulous mannequin we have today in Doppelganger form :)
image

File: unpublishedScripts/marketplace/doppleganger-attachments/app-doppleganger-attachments.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+UserActivityLogger.logAction('doppleganger_app_load');
+doppleganger.activeChanged.connect(function(active, reason) {
+    if (active) {
+        UserActivityLogger.logAction('doppleganger_enable');
+    } else {
+        if (reason === 'stop') {
+            // user intentionally toggled the doppleganger
+            UserActivityLogger.logAction('doppleganger_disable');
+        } else {
+            print('doppleganger stopped:', reason);
```
Comment (PhilipRosedale) [inline unpublishedScripts/marketplace/doppleganger-attachments/app-doppleganger-attachments.js]: Quiet print debugs throughout.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1057/HighFidelity-Beta-PR10743-d6eb77e815147b818fe3129fb6ea5cb2f827d98f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1057/HighFidelity-Beta-PR10743-d6eb77e815147b818fe3129fb6ea5cb2f827d98f.exe

**Pull request merged**